src/metaswitch/ellis/prov_tools/list_users.py

```diff
@@ -0,0 +1,108 @@
+# details. You should have received a copy of the GNU General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+# The author can be reached by email at clearwater@metaswitch.com or by
+# post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+#
+# Special Exception
+# Metaswitch Networks Ltd  grants you permission to copy, modify,
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+import sys
+import logging
+import argparse
+import time
+
+from tornado.web import HTTPError
+from metaswitch.ellis import settings
+from metaswitch.ellis.prov_tools import utils
+
+_log = logging.getLogger();
+
+def pace_wrapper(iterator, pace=10):
+    next_time = time.time()
+    for item in iterator:
+        now_time = time.time()
+        if now_time < next_time:
+            time.sleep(next_time - now_time)
+        else:
+            next_time = now_time
+        next_time += 1.0 / pace
+        yield item
+
+def main():
+    parser = argparse.ArgumentParser(description="List users")
+    parser.add_argument("-k", "--keep-going", action="store_true", dest="keep_going", help="keep going on errors")
+    parser.add_argument("--hsprov", metavar="IP:PORT", action="store", help="IP address and port of homestead-prov")
+    parser.add_argument("--full", action="store_true", help="displays full information for each user")
+    parser.add_argument("--pace", action="store", help="sets the target number of users to list per second")
+    parser.add_argument("-f", "--force", action="store_true", dest="force", help="forces specified pace")
+    args = parser.parse_args()
+
+    utils.setup_logging()
+    settings.HOMESTEAD_URL = args.hsprov or settings.HOMESTEAD_URL
+    default_pace = 5 if args.full else 500
+    pace = int(args.pace) if args.pace else default_pace
+
+    # Check the pace and get confirmation if it's too high
+    if pace > default_pace and not args.force:
+        if args.full:
+            print("Pace %d greater than recommended value (%d) when --full specified" % (pace, default_pace))
+        else:
+            print("Pace %d greater than recommended value (%d)" % (pace, default_pace))
+        print("This may impact call processing!  Are you sure?")
```

Comment:
I think we go for "Question? [y/N]" elsewhere, so maybe we should do that here for consistency of user interface.

Admittedly we only do "[y/N]" elsewhere because I wrote that code last week, so I can't really make an argument from long tradition if you want to standardise on "Type <yes> below".

Maybe we should have an "ask a question and exit on no" function in python-common?